File: app/src/main/java/com/tuvakov/zeyoube/android/utils/DateTimeUtils.java
package com.tuvakov.zeyoube.android.utils;

import org.threeten.bp.Instant;

import java.util.concurrent.TimeUnit;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class DateTimeUtils {

    private static final long SECOND_IN_MILLIS = 1000;
    private static final long MINUTE_IN_MILLIS = SECOND_IN_MILLIS * 60;
    private static final long HOUR_IN_MILLIS = MINUTE_IN_MILLIS * 60;
    private static final long DAY_IN_MILLIS = HOUR_IN_MILLIS * 24;

    @Inject
    public DateTimeUtils() { }

    public long getUtcEpoch() {
        return Instant.now().toEpochMilli();
    }

    public long getUtcEpochNDaysAgo(int n) {
        Instant now = Instant.now();
        return now.minusMillis(TimeUnit.DAYS.toMillis(n)).toEpochMilli();
    }

    public boolean hasDayPassed(long since) {
        return getUtcEpoch() - since >= DAY_IN_MILLIS;
    }
}

File: app/src/main/java/com/tuvakov/zetube/android/utils/DateTimeUtils.kt
package com.tuvakov.zetube.android.utils

import com.tuvakov.zetube.android.utils.SyncUtils.Companion.SYNC_INTERVAL_HOURS
import org.threeten.bp.Instant
import java.util.concurrent.TimeUnit
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class DateTimeUtils @Inject constructor() {
    val utcEpoch: Long
        get() = Instant.now().toEpochMilli()

    fun getUtcEpochNDaysAgo(n: Int): Long {
        val now = Instant.now()
        return now.minusMillis(TimeUnit.DAYS.toMillis(n.toLong())).toEpochMilli()
    }

    fun isSyncAllowed(since: Long): Boolean {
        return utcEpoch - since >= HOUR_IN_MILLIS * SYNC_INTERVAL_HOURS
    }

    companion object {
        private const val SECOND_IN_MILLIS: Long = 1000
        private const val MINUTE_IN_MILLIS = SECOND_IN_MILLIS * 60
        private const val HOUR_IN_MILLIS = MINUTE_IN_MILLIS * 60
    }
}